x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/RemoteAccessAuthentication.java

```diff
@@ -161,6 +157,10 @@ public RoleDescriptorsBytes(StreamInput streamInput) throws IOException {
             this(streamInput.readBytesReference());
         }
 
+        public void writeTo(StreamOutput out) throws IOException {
```

Comment:
Your suggestion makes total sense to me. I was wondering along the same line as well. Since we now explicilty handles de/serialization for RoleDescriptorBytes, it does not benefit much more from being a subclass of BytesRefernce anymore. I removed the inheritance and made it implement Writeable as suggested. Please take a look at the latest change 04a4fc8149ddfe6aa15468177fb1e4e72b255188 and see if it meets your expectation. Thanks!

Reply:
Exactly what I had in mind. Thanks!